cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/api/artifact/ArtifactVersionTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.artifact;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests the parsing of artifact version.
+ */
+public class ArtifactVersionTest {
+
+  @Test
+  public void testValidVersions() {
+    // since a version's suffix can not have numbers in it, simply the number 1 is the version
+    String versionString = "cdap-etl-batch-3.5-SNAPSHOT1";
```

Comment:
good point. this should be an invalid version. We need a as the prefix in the regex matcher, then?